src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -214,5 +227,13 @@ protected virtual void Dispose(bool disposing)
             ConventionBasedOperationProvider.ApplyTo(builder, targetType);
             ConventionBasedEntitySetFilter.ApplyTo(builder, targetType);
         }
+
+        // Registered as a scoped service so that IApi and ApiContext could be exposed as scoped service.
+        // If a descendant class wants to expose these 2 services in another way, it must ensure they could be
+        // resolved after CreateApiContext call.
+        private class ApiScope
```

Comment:
Previously I thought this was the scope class but I found it doesn't inherit `IServiceScope` and its only usage is to keep the instance of an `Api` class which will be later used in service factory methods. So I suggest we use a different name for this class like `ApiKeeper`, `ApiHolder`, or else to avoid any confusion with the context scope.